apps/generator/docs/generator-template.md

```diff
@@ -643,10 +643,558 @@ Temperature drop detected 66943992 sent to temperature/dropped
 Temperature rise detected 66943992 sent to temperature/risen
 ```
 
+## Creating a Java MQTT client template
+
+Looking for more?
+
+This section guides you through creating a flexible MQTT-supported template that will generate a **Java** client from the template and the AsyncAPI document referenced above.
+The following steps are similar to the [initial steps](https://www.asyncapi.com/docs/tools/generator/generator-template#overview-of-steps) but with a few differences. This will help developers practice with AsyncAPI generator tool using statically typed language **Java**.
+
+## Java Prerequisites
+
+To run it, ensure you have Java JDK 8 or higher, Gradle, and the AsyncAPI generator.
+- **Gradle** - Get gradle at https://gradle.org/install/ 
+- **JDK** - Get jdk at https://www.oracle.com/ca-en/java/technologies/downloads/
+
+## Overview of Java Template
+
+In this section, you'll:
+
+1. Create new directory to run Java code.
+2. Create the Java client.
+3. Output Java template code.
+4. Create more channels
+
+### 1. Create new directory to run Java code
+
+Create a new directory called **java-mqtt-client-template** at the root of your project. This is where all your java templating work will go.
+
+Once that is done you should create some new sub-directories to begin building your Java client.
+  1. Create a new subdirectory called `src`
+  2. Change into `src` and create two new subdirectories: `fixtures` and `main/java/template`.
+  3. Create a file named asyncapi.yml in your fixtures directory and paste the asyncapi.yml document mentionned [above](https://www.asyncapi.com/docs/tools/generator/generator-template#background-context) into it.
+  4. Create a new file named **package.json** in your **java-mqtt-client-template** directory. This file is used to define the **dependencies** for your template.
+  5. Create a new file called **build.gradle** in your **java-mqtt-client-template** directory. This file is used to build your generated java code for your template.
+  6.  Create a new file named **index.js** in your **template** directory. This file is used to define the **logic** for your template.
+
+Now your directory should look like this:
+
+```
+java-mqtt-client-template 
+├── src
+|   └── fixtures
+|       └── asyncapi.yml
+│   └── main/java
+├── template
+|   └── index.js
+└── package.json
+└── build.gradle
+```
+
+
+Note: the **client.java** code must be in your **src/main/java** directory, or else gradle won't build your applicaiton.
+
+### java - package.json file
+Add the following code snippet to your package.json file:
```

Comment:
Hey @ssala034 , for the file names, you’re mixing the use of backticks`` and asterisks **. Please use `` to reference files and directories in the docs as stipulated in the formatting guide here -> https://www.asyncapi.com/docs/community/styleguide/formatting

> Use code style for filenames, directories, and paths. For example: Go to the `/docs/tutorials` directory.